src/Integration/Binding/IProjectBinder.cs

```diff
@@ -29,5 +31,6 @@ namespace SonarLint.VisualStudio.Integration.Binding
     internal interface IProjectBinder
     {
         bool IsBound(BindingConfiguration binding, Project project);
+        Action GetBindAction(IBindingConfigFile configFile, Project project, CancellationToken cancellationToken);
```

Comment:
I'm not a big fan of this method. It essentially returns a handle to ProjectBindingOperation.Commit.
At first I created PrepareBind and CommitBind methods, but that required to store the ProjectBindingOperation as a member of IProjectBinder and I was not a fan of that.
Let me know what you think.

Reply:
Yep, returning an `Action` feels wrong. It's like using `object` as a data type - it doesn't convey any semantic information. It would be better to have an interface/delegate to encapsulate the "commit binding" concept.

Probably the simplest thing to do at this point would be declare a delegate and use that as the return type instead e.g. `public delegate void BindProject();`
That would be simpler than adding a new interface with a single operation.

Essentially, at this point we want to get the top-down API right. We can hide existing implementation classes like `ProjectBindingOperation` behind then new API and refactor them later if necessary.